### DISCUSSION TITLE
"Depends"/"Security" in every single endpoint/route?

### DISCUSSION THREAD

**Discussion by MaWe4585** [fastapi/fastapi#6342] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Depends, FastAPI, Security
from fastapi.security import OpenIdConnect, OAuth2PasswordBearer, OAuth2AuthorizationCodeBearer

app = FastAPI()

oauth2_scheme = OAuth2AuthorizationCodeBearer(
    authorizationUrl="http://localhost:9000/auth",
    tokenUrl="http://localhost:9000/token",
    refreshUrl="http://localhost:8000/auth/callback",
    scopes={"openid":"auth with openid","email":"Access to the mail","profile":"access user information"}
)

@app.get("/")
def get_root(token: str = Security(oauth2_scheme)):
    return { "message":"success" }

@app.get("/auth/callback")
def get_root(token: str = Security(oauth2_scheme)):
    return { "message":"callback success" }
```

### Description

Im new to FastAPI and Python and I'm trying to add OIDC / OAuth2 to an API.

Currently I'm wondering, if it is really necessary to add the Security Parameter to every single endpoint?
In the end I will have around 200 endpoints. 
The whole API - every single call - requires the user to be authenticated.
Either because the data is just sensitive or because it belongs to the user and he is only allowed to access his own data.

Do you know of a way where I can add this to one place only so that I won't accidentally forget to secure one endpoint?
I still need to be able to access the user information from within each endpoint..

### Operating System

Windows

### FastAPI Version

0.74.1

### Python Version

3.9.10

**Comment by n8sty** [comment, marked as answer]
I think you're looking for the `dependencies` parameter of the `fastapi.FastAPI` or `fastapi.APIRouter` classes which allows you to apply a dependency to all routes within that object

**Comment by MaWe4585** [comment]
I think that was the hint I needed! So basically a global dependency is what I need! https://fastapi.tiangolo.com/tutorial/dependencies/global-dependencies/

**Comment by n8sty** [comment]
Great! Please close if your issue is resolved

**Comment by MaWe4585** [comment]
Thanks a lot!
